src/libcollections/vec.rs

```diff
@@ -1106,12 +1106,36 @@ impl<T: Clone> Vec<T> {
         let len = self.len();
 
         if new_len > len {
-            self.extend(repeat(value).take(new_len - len));
+            self.extend_elem(new_len - len, value);
```

Comment:
Could you add a comment here to the effect of why we're not using `extend` + `repeat`?